src/ansys/simai/core/data/model_configuration.py

```diff
@@ -440,7 +441,13 @@ def _to_payload(self):
         if self.global_coefficients is not None:
             gcs = [asdict(gc) for gc in self.global_coefficients]
 
-        surface_pp_input_fld = self.pp_input.surface or []
+        surface_pp_input_fld = []
+        if self.pp_input.surface is not None:
+            sample_surface = dict_get(sample_metadata, "surface", {})
+            suface_fields = dict_get(sample_surface, "fields", default=[])
```

Comment:
```suggestion
            suface_fields = dict_get(sample_metadata, "surface", "fields", default=[])
```

Reply:
yup, and actually the previous assignment was packing `{}` too, that was creating problems. It's set now.